docs/fundamentals/code-analysis/quality-rules/ca1839.md

```diff
@@ -0,0 +1,106 @@
+---
+title: "CA1839: Use 'Environment.ProcessPath'"
+description: "'Environment.ProcessPath' is simpler and faster than 'Process.GetCurrentProcess().MainModule.FileName'."
+ms.date: 11/23/2021
+ms.topic: reference
+f1_keywords:
+  - "CA1839"
+helpviewer_keywords:
+  - "CA1839"
+author: buyaa-n
+dev_langs:
+- CSharp
+- VB
+---
+
+# CA1839: Use `Environment.ProcessPath` instead of `Process.GetCurrentProcess().MainModule.FileName`
+
+| | Value |
+|-|-|
+| **Rule ID** |CA1839|
+| **Category** |[Performance](performance-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+Using `Process.GetCurrentProcess().MainModule.FileName` for getting the path to the file that launched the process instead of `Environment.ProcessPath`.
```

Comment:
```suggestion
Using `Process.GetCurrentProcess().MainModule.FileName` for getting the path to the file that launched the process instead of <xref:System.Environment.ProcessPath?displayProperty=nameWithType>.
```